### PR TITLE
controllers: move status updates into the reconcile loop

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas, @landism,

Please review the following commits I made in branch nicks/imager:

878270caabcd9c2990e768bcf7220d4c442b3c99 (2022-04-15 11:33:28 -0400)
controllers: move status updates into the reconcile loop

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/engine/buildcontrol/image_build_and_deployer_test.go [hunk 1]
```diff
@@ -869,6 +870,8 @@ func TestDockerBuildStatus(t *testing.T) {
 	_, err = f.BuildAndDeploy(BuildTargets(manifest), store.BuildStateSet{})
 	require.NoError(t, err)
 
+	f.ibd.dr.Reconcile(f.ctx, ctrl.Request{NamespacedName: nn})
```
Comment (milas) [inline internal/engine/buildcontrol/image_build_and_deployer_test.go:873]: Would it make sense to move this into `ibdFixture::BuildAndDeploy`?

**Review by milas — APPROVED** [review]

File: internal/controllers/core/cmdimage/reconciler.go [hunk 2]
```diff
@@ -76,19 +97,97 @@ func (r *Reconciler) ForceApply(
 	// if you want the live container to keep receiving updates
 	// while an image build is going on in parallel.
 	startTime := apis.NowMicro()
-	MaybeUpdateStatus(ctx, r.client, iTarget, ToBuildingStatus(iTarget, startTime))
+	nn := types.NamespacedName{Name: iTarget.CmdImageName}
+	r.setImageStatus(nn, ToBuildingStatus(iTarget, startTime))
+	r.requeuer.Add(nn)
+	defer r.requeuer.Add(nn)
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/controllers/core/cmdimage/reconciler.go:106]: yep! left a comment

File: internal/engine/buildcontrol/image_build_and_deployer_test.go (hunk 1 above)
Comment (nicks) [reply internal/engine/buildcontrol/image_build_and_deployer_test.go]: done

File: internal/controllers/core/cmdimage/reconciler.go [hunk 3]
```diff
@@ -76,19 +97,97 @@ func (r *Reconciler) ForceApply(
...
+		return nil
+	}
+
+	update := obj.DeepCopy()
+	update.Status = *(newStatus.DeepCopy())
+
+	return r.client.Status().Update(ctx, update)
 }
 
 func (r *Reconciler) CreateBuilder(mgr ctrl.Manager) (*builder.Builder, error) {
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/controllers/core/cmdimage/reconciler.go:196]: done

**Pull request merged**